---
title: "Ch2Blodgett_RawData"
output: html_document
date: "2024-04-17"
---
# LOAD PACKAGES
```{r}
library(tidyverse)
library(dplyr)
```

# VIROME ABUNDANCE TABLE
```{r}
# LOAD ABUNDANCE DATA FOR ENTIRE DATASET
otu=read_delim("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch2Blodgett/R_Figures/input data/all.good.75.tmean.tsv", delim = "\t", col_names = TRUE)
# prep data frame
otu <- col_to_row(otu, "Contig", "contignames")
# edit column names 
colnames(otu) <- gsub(pattern=".vib.sI.Trimmed.Mean", replacement = "", x=colnames(otu))
# remove two samples that are not necessary
otu$N1_S61_L004 <- NULL
otu$N2_S62_L004 <- NULL
# save RDS file
saveRDS(otu, "C:/Users/segeo/Box Sync/EmersonLab/Research/Ch2Blodgett/R_Figures/rds/blodgett_all_otu.RDS")
```

# VIROME COUNT TABLE
```{r}
# LOAD COUNT TABLE
otu.count=read_delim("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch2Blodgett/R_Figures/input data/all.count.tsv", delim = "\t", col_names = TRUE)
otu.count <- col_to_row(otu.count, "Contig", "contignames")
colnames(otu.count) <- gsub(pattern=".vib.sI.Read.Count", replacement = "", x=colnames(otu.count))
otu.count=otu.count%>%
  select(-N1_S61_L004, -N2_S62_L004)
# save RDS file
saveRDS(otu.count, "C:/Users/segeo/Box Sync/EmersonLab/Research/Ch2Blodgett/R_Figures/rds/blodgett.otu.count.RDS")
```

# VIROME METADATA
```{r}
# heat metadata
heatmetadata=read_delim("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch1SoilHeating/R_Figures/input data/heat_metadata.txt", delim = "\t", col_names = TRUE)
colnames(heatmetadata)[colnames(heatmetadata) == "Treatment"] <- "Experiment"
heatmetadata <- heatmetadata %>%
  mutate(Treatment = case_when(
    Temp == "Control" ~ "Field",
    Temp == "10C" ~ "Control",
    Temp == "30C" ~ "30ºC",
    Temp == "60C" ~ "60ºC", 
    Temp == "90C" ~ "90ºC")) %>%
  mutate(TempGroup = case_when(
    Treatment == "Field" ~ "Field/Control/30ºC", 
    Treatment == "Control" ~ "Field/Control/30ºC", 
    Treatment == "30ºC" ~ "Field/Control/30ºC", 
    Treatment == "60ºC" ~ "60ºC", 
    Treatment == "90ºC" ~ "90ºC")) %>%
  mutate(Virome = case_when(
    DNase == "NoDNase" ~ "NonDNase",
    DNase == "DNase" ~ "DNase")) %>%
  mutate(Depth = case_when(
    Horizon == "O" ~ "0-3cm",
    Horizon == "A" ~ "3-6cm")) %>%
  mutate(DNase = case_when(
    DNase == "NoDNase" ~ "-",
    DNase == "DNase" ~ "+"))
heatmetadata$Plot <- 'U1'
heatmetadata$Date <- '4/18/2022'
heatmetadata <- heatmetadata %>%
  mutate(Treatment = case_when(
    Temp == "Control" ~ "Heat_Field",
    Temp == "10C" ~ "Heat_Control",
    Temp == "30C" ~ "30ºC",
    Temp == "60C" ~ "60ºC", 
    Temp == "90C" ~ "90ºC"))
heatmetadata <- heatmetadata %>%
  mutate(Disturbance = case_when(
    Treatment == "Heat_Field" ~ "No",
    Treatment == "Heat_Control" ~ "No",
    Treatment == "30ºC" ~ "Yes_30ºC",
    Treatment == "60ºC" ~ "Yes_60ºC", 
    Treatment == "90ºC" ~ "Yes_90ºC"))
heatmetadata$SubPlot <- 'NA'
heatmetadata$Burn <- 'NA'
heatmetadata <- select(heatmetadata, -Horizon, -Temp, -Experiment, -TempGroup)
heatmetadata <- select(heatmetadata, -Virome)

# field metadata
fieldmetadata=read_delim("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch2Blodgett/R_Figures/input data/blodgett_pcoa.tsv", delim = "\t", col_names = TRUE)
fieldmetadata <- select(fieldmetadata, -pcoa1, -pcoa2, -SampleID)
fieldmetadata <- fieldmetadata %>%
  mutate(Treatment = case_when(
    Treatment == "Control" ~ "Prescribed_Control",
    Treatment == "Burn" ~ "Prescribed_Burn"))
fieldmetadata$DNase <- '+'
fieldmetadata$Replicate <- 'NA'
fieldmetadata <- fieldmetadata %>%
  mutate(Disturbance = case_when(
    Disturbance == "yes" ~ "Yes_Burn", 
    Disturbance == "no" ~ "No"))
colnames(fieldmetadata)[1] <- "ID"

blodgett_data <- rbind(heatmetadata, fieldmetadata)

# make sure IDs are in the same order as the column names of the OTU dataframe
column_names <- names(otu)
blodgett_data <- blodgett_data[match(column_names, blodgett_data$ID), ]

# add environmental cluster data
k_cluster_env=readRDS("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch2Blodgett/R_Figures/rds/k_cluster_env.RDS")
# change column name of SampleID to ID
colnames(k_cluster_env)[colnames(k_cluster_env) == "SampleID"] <- "ID"
blodgett_data = left_join(blodgett_data, k_cluster_env, by="ID")
blodgett_data <- blodgett_data %>%
  mutate(Cluster = as.character(Cluster))

# make sure IDs are identical in count and metadata tables
identical(colnames(otu.count), blodgett_data$ID)

saveRDS(blodgett_data, "C:/Users/segeo/Box Sync/EmersonLab/Research/Ch2Blodgett/R_Figures/rds/blodgett_all_metadata.RDS")

# DNA Yields
blodgett_dnayields=read_delim("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch2Blodgett/R_Figures/input data/blodgettSSSA.tsv", delim = "\t", col_names = TRUE)
blodgett_dnayields <- blodgett_dnayields %>%
  mutate(Date = case_when(
    Date == "4/5/2021" ~ "T1",
    Date == "4/18/2021" ~ "T2", 
    Date == "4/27/2021" ~ "T3", 
    Date == "5/8/2021" ~ "T4",
    Date == "6/21/2021" ~ "T5"))
saveRDS(blodgett_dnayields, "C:/Users/segeo/Box Sync/EmersonLab/Research/Ch2Blodgett/R_Figures/rds/blodgett_dnayields.RDS")

```

# 16S DATA
```{r}
# LOAD & CLEAN TAXONOMY DATA
tax_all <- readRDS("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch2Blodgett/R_Figures/rds/tax_all.RDS")
# Change NAs in taxonomy to "unknown"
tax_all$Family[is.na(tax_all$Family)] <- "Unknown Family"
tax_all$Order <- factor(tax_all$Order, levels = c(levels(tax_all$Order), "Unknown Order"))
tax_all$Order[is.na(tax_all$Order)] <- "Unknown Order"
tax_all$Class <- factor(tax_all$Class, levels = c(levels(tax_all$Class), "Unknown Class"))
tax_all$Class[is.na(tax_all$Class)] <- "Unknown Class"
tax_all$Phylum <- factor(tax_all$Phylum, levels = c(levels(tax_all$Phylum), "Unknown Phylum"))
tax_all$Phylum[is.na(tax_all$Phylum)] <- "Unknown Phylum"
# remove mitochondria and chloroplast
criteria <- !(tax_all$Family == "Mitochondria" | tax_all$Order == "Chloroplast")
tax_filtered <- tax_all[criteria, ]
# change Chloroflexi to Chloroflexota
tax_filtered <- tax_filtered %>%
  mutate(Phylum = as.character(Phylum)) %>%
  mutate(Phylum = ifelse(Phylum == "Chloroflexi", "Chloroflexota", Phylum))
unique(tax_filtered$Phylum)

saveRDS(tax_filtered, "C:/Users/segeo/Box Sync/EmersonLab/Research/Ch2Blodgett/R_Figures/rds/tax_filtered.RDS")

# LOAD & CLEAN ABUNDANCE DATA
asv_all <- readRDS("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch2Blodgett/R_Figures/rds/asv_all.RDS")
asv_all <- as.data.frame(asv_all)
asv_all$OTU_ID <- rownames(asv_all)
# make sure that asv_all is in the same order as tax_all by OTU_ID (return = TRUE)
check_column <- "OTU_ID"
all(asv_all[[check_column]] == tax_all[[check_column]])
rownames(asv_all) <- NULL
#remove ASVs that were assigned Mitochondria or Chloroplast
asv_filtered <- asv_all[!(tax_all$Family=="Mitochondria"),]
tax_no_mitochondria <- tax_all[!(tax_all$Family=="Mitochondria"),]
asv_filtered <- asv_filtered[!(tax_no_mitochondria$Order=="Chloroplast"),]

# SELECT FOR Blodgett DATASET
#load 16S mapping data
all_map=read_delim("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch2Blodgett/R_Figures/input data/16S_ID.txt", delim = "\t", col_names = TRUE)
#change SampleID to ID
colnames(all_map)[colnames(all_map) == "SampleID"] <- "ID"
#subset blodgett 16S data
blodgett.asv <- asv_filtered[,all_map$Location=="Blodgett"]
blodgett.asv <- subset(blodgett.asv, select = -c(FIRE061, FIRE060))

# PREP ABUNDANCE DATA
#do some other things to prep data frame
rownames(blodgett.asv) <- blodgett.asv$OTU_ID
blodgett.asv$OTU_ID <- NULL
# remove rows with just zeroes (39,258)
blodgett.asv <- blodgett.asv[rowSums(blodgett.asv>0) >0,]
# remove singletons (9,210) 
blodgett.asv <- rmv_sngl(blodgett.asv)

saveRDS(blodgett.asv, "C:/Users/segeo/Box Sync/EmersonLab/Research/Ch2Blodgett/R_Figures/rds/blodgett.asv.RDS")

# 
write.csv(blodgett.asv, "C:/Users/segeo/Box Sync/EmersonLab/Research/Ch2Blodgett/16S_rRNA_ASV_Abundance_Table.csv")

#load mapping data to subset within blodgett dataset 
#subset mapping data
prescribed_map <- read_delim("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch2Blodgett/R_Figures/input data/prescribed_16S_map.txt", delim = "\t", col_names = TRUE)
prescribed_map <- prescribed_map %>%
  mutate(Treatment = case_when(
    Treatment == "Control" ~ "Prescribed_Control",
    Treatment == "Burn" ~ "Prescribed_Burn"))
prescribed_map$SampleType <- 'Total'
prescribed_map$Replicate <- 'NA'
prescribed_map <- prescribed_map %>%
  mutate(Disturbance = case_when(
    Disturbance == "Yes" ~ "Yes_Burn", 
    Disturbance == "No" ~ "No"))
heat_map=read_delim("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch2Blodgett/R_Figures/input data/heat16S_mapwrep.txt", delim = "\t", col_names = TRUE)
heat_map$Plot <- 'U1'
heat_map$Date <- '4/18/2022'
heat_map <- heat_map %>%
  mutate(Treatment = case_when(
    Temp == "Control" ~ "Heat_Field",
    Temp == "10C" ~ "Heat_Control",
    Temp == "30C" ~ "30ºC",
    Temp == "60C" ~ "60ºC", 
    Temp == "90C" ~ "90ºC"))
heat_map <- subset(heat_map, select = -Temp)
heat_map <- heat_map %>%
  mutate(Depth = case_when(
    Horizon == "O" ~ "0-3cm",
    Horizon == "A" ~ "3-6cm"))
heat_map <- subset(heat_map, select = -Horizon)
heat_map <- heat_map %>%
  mutate(SampleType = case_when(
    Virome == "Bulk" ~ "Total",
    Virome == "Virome" ~ "Virome"))
heat_map <- subset(heat_map, select = -Virome)
heat_map <- heat_map %>%
  mutate(Disturbance = case_when(
    Treatment == "Heat_Field" ~ "No",
    Treatment == "Heat_Control" ~ "No",
    Treatment == "30ºC" ~ "Yes_30ºC",
    Treatment == "60ºC" ~ "Yes_60ºC", 
    Treatment == "90ºC" ~ "Yes_90ºC"))
heat_map <- subset(heat_map, select = -Location)
heat_map <- subset(heat_map, select = -Experiment)
heat_map$SubPlot <- 'NA'
heat_map$Burn <- 'NA'
blodgett_map <- rbind(heat_map, prescribed_map)
blodgett_map <- blodgett_map %>%
  mutate(Date = case_when(
    Date == "4/5/2021" ~ "4/5/2021",
    Date == "4/18/2021" ~ "4/18/2021", 
    Date == "4/27/2021" ~ "4/27/2021", 
    Date == "5/8/2021" ~ "5/8/2021",
    Date == "6/21/2021" ~ "6/20/2021", 
    Date == "4/18/2022" ~ "4/18/2022"))

# make sure mapping file has sample names (rownames) in ascending order
blodgett_map <- blodgett_map[order(blodgett_map$ID), ]

# make sure IDs are identical in abundance and metadata tables
identical(colnames(blodgett.asv), blodgett_map$ID)

saveRDS(blodgett_map, "C:/Users/segeo/Box Sync/EmersonLab/Research/Ch2Blodgett/R_Figures/rds/blodgett_map.RDS")

# write.table(x=blodgett_map, file = "Blodgett_16S_Metadata.tsv", quote=FALSE, sep="\t")
```
