---
title: "Ch3Wildfire_16S"
output: html_document
date: "2024-06-27"
---
```{r}
# LOAD PACKAGES
library(tidyverse)
library(vegan)
library(dplyr)
library(ggplot2)
library(factoextra)
library(gridExtra)
library(cowplot)
library(ComplexUpset)

# Load RDS files
lnu.asv <- readRDS("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch3Wildfire/R_Figures/rds/lnu.asv.RDS")

#write.csv(lnu.asv, "C:/Users/segeo/Box Sync/EmersonLab/Research/Ch3Wildfire/Journal Submission/lnu.asv.csv", row.names = TRUE)

tax_filtered <- readRDS("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch3Wildfire/R_Figures/rds/tax_filtered.RDS")
wildfire_asv_map <- readRDS("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch3Wildfire/R_Figures/rds/wildfire_asv_map.RDS")
```

# PCoA: ALL DATA ... do this before taxonomy 
```{r}
# create rarefaction curve to determine rarefaction threshold for abundance table
# don't need to run the below code every time, only the first time to figure out what to rarefy to
#asv.rarecurve <- rarecurve(t(lnu.asv), step = 100) 
#x <- map_dfr(asv.rarecurve, bind_rows) %>% bind_cols(colnames(lnu.asv),.)
#x <- as.data.frame(x)
#row.names(x) <- x[, 1]

#based on x dataframe, choose sample size to rarefy to
set.seed(24)
asv.rare <- t(rrarefy(t(lnu.asv), sample = 17700))

# remove singletons --> 4,933 contigs
asv.rare <- data.frame(asv.rare) %>%
  rmv_sngl()

#bray-curtis dissimilarity distance matrix 
asv.dist=vegdist(t(asv.rare), method = "bray")
asv.pcoa <- pcoa(asv.dist)
asv.pcoa.points <- pcoa.points(asv.pcoa)
asv.map <- pcoa.map.asv(asv.pcoa.points, wildfire_asv_map)
variance(asv.pcoa, 1)
variance(asv.pcoa, 2)

ggplot(asv.map, aes(x=pcoa1, y=pcoa2, color=Treatment, shape = Habitat))+
  geom_point(size = 3) +
  labs(x = "PCo1 (20.89% Variance Explained)", 
       y = "PCo2 (11.47% Variance Explained)",
       title = "All LNU 16S")+
  theme(axis.title = element_text(size = 18)) +
  theme(legend.text = element_text(size = 18), legend.position = "right") +
  theme(legend.title = element_text(size = 18)) +
  theme(axis.text = element_text(size = 18)) +
  scale_color_manual(values=pf_treatment_color, limits=pf_treatment_order) +
  theme(panel.background=element_rect(fill="white"), 
      panel.grid.major = element_line(color = "lightgrey"), 
      panel.grid.minor = element_line(color = "lightgrey"))

ggsave("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch3Wildfire/R_Figures/pdf/PCoA/prokaryotes/all_LNU_ASV_pcoa_treatment_habitat.pdf", width = 200, height = 125, units = "mm", dpi = 500)

pcoa.treatment.habitat(asv.map, x_col = pcoa1, y_col = pcoa2, color_col= Treatment, shape_col = Habitat, x_label = "20.89% Variance Explained", y_label = "11.47% Variance Explained", title = "All LNU 16S rRNA ASV Profile", filename = "C:/Users/segeo/Box Sync/EmersonLab/Research/Ch3Wildfire/R_Figures/pdf/PCoA/prokaryotes/all_LNU_ASV_pcoa_treatment_habitat.pdf")

pcoa.treatment.time(asv.map, x_col = pcoa1, y_col = pcoa2, color_col= Treatment, shape_col = TimePoint, x_label = "20.89% Variance Explained", y_label = "11.47% Variance Explained", title = "All LNU 16S rRNA ASV Profile", filename = "C:/Users/segeo/Box Sync/EmersonLab/Research/Ch3Wildfire/R_Figures/pdf/PCoA/prokaryotes/all_LNU_ASV_pcoa_treatment_time.pdf")

pcoa.habitat.location(asv.map, x_col = pcoa1, y_col = pcoa2, color_col= Habitat, shape_col = Location, x_label = "20.89% Variance Explained", y_label = "11.47% Variance Explained", title = "All LNU 16S rRNA ASV Profile", filename = "C:/Users/segeo/Box Sync/EmersonLab/Research/Ch3Wildfire/R_Figures/pdf/PCoA/prokaryotes/all_LNU_ASV_pcoa_habitat_location.pdf")

# run PERMANOVAs
variables <- c("Habitat", "Treatment", "Location")
adonis.results <- run_adonis(variables, asv.dist, asv.map)
adonis.table <- do.call(rbind, adonis.results)

```

# Subset data
# Only T3-T5 for burn vs. control direct comparisons (Chaparral)
```{r}
bc.ch.asv <- lnu.asv[, wildfire_asv_map$TimePoint %in% c("T3", "T4", "T5") & wildfire_asv_map$Habitat=="Chaparral"] 

# create rarefaction curve to determine rarefaction threshold for abundance table
# don't need to run the below code every time, only the first time to figure out what to rarefy to
#asv.rarecurve <- rarecurve(t(bc.ch.asv), step = 100) 
#x <- map_dfr(asv.rarecurve, bind_rows) %>% bind_cols(colnames(bc.ch.asv),.)
#x <- as.data.frame(x)
#row.names(x) <- x[, 1]

#based on x dataframe, choose sample size to rarefy to
set.seed(24)
bc.ch.asv.rare <- t(rrarefy(t(bc.ch.asv), sample = 17700))

# remove singletons --> 2,475 contigs
bc.ch.asv.rare <- data.frame(bc.ch.asv.rare) %>%
  rmv_sngl()

bc.ch_map <- subset(wildfire_asv_map, TimePoint %in% c("T3", "T4", "T5") & Habitat == "Chaparral")

# make sure IDs are identical in abundance and metadata tables
identical(colnames(bc.ch.asv), bc.ch_map$ASV_ID)
```

# Only T3-T5 for burn vs. control direct comparisons (Woodland)
```{r}
bc.wo.asv <- lnu.asv[, wildfire_asv_map$TimePoint %in% c("T3", "T4", "T5") & wildfire_asv_map$Habitat=="Woodland"] 

# create rarefaction curve to determine rarefaction threshold for abundance table
# don't need to run the below code every time, only the first time to figure out what to rarefy to
#asv.rarecurve <- rarecurve(t(bc.wo.asv), step = 100) 
#x <- map_dfr(asv.rarecurve, bind_rows) %>% bind_cols(colnames(bc.wo.asv),.)
#x <- as.data.frame(x)
#row.names(x) <- x[, 1]

#based on x dataframe, choose sample size to rarefy to
set.seed(24)
bc.wo.asv.rare <- t(rrarefy(t(bc.wo.asv), sample = 18700))

# remove singletons --> 2,515 contigs
bc.wo.asv.rare <- data.frame(bc.wo.asv.rare) %>%
  rmv_sngl()

bc.wo_map <- subset(wildfire_asv_map, TimePoint %in% c("T3", "T4", "T5") & Habitat == "Woodland")

# make sure IDs are identical in abundance and metadata tables
identical(colnames(bc.wo.asv), bc.wo_map$ASV_ID)
```

# Taxonomy Stacked Bar Plot: ALL DATA (2021 + 2022)
```{r}
# add OTU_ID back onto asv.rare
asv.rare$OTU_ID <- rownames(asv.rare)

# Join taxonomy and abundance data together
tax_lnu <- left_join(asv.rare, tax_filtered, by = "OTU_ID")
# make sure asv.rare is in the same order as tax_lnu (return = TRUE)
check_column <- "OTU_ID"
all(asv.rare[[check_column]] == tax_lnu[[check_column]])
unique(tax_lnu$Phylum)

# pivot table and group by Phylum
tax_lnu.long <- pivot_longer(tax_lnu, cols=1:87, names_to = "ASV_ID", values_to = "counts")
lnu.phylum.long <- group_by(tax_lnu.long, Phylum, ASV_ID) %>%
  dplyr::summarize(counts=sum(counts))

# collapse phyla into "other" category if represents less than 2%, which is 354 (rarefied to 17,700)
lnu.phylum.long$Phylum <- as.character(lnu.phylum.long$Phylum)
lnu.phylum.long <- lnu.phylum.long %>%
  group_by(Phylum) %>%
  mutate(NewPhylum = ifelse(all(counts < 354), 'Other', Phylum)) %>%
  ungroup()
# find out which phyla are represented (used for making legend)
unique(lnu.phylum.long$NewPhylum)

saveRDS(lnu.phylum.long, "C:/Users/segeo/Box Sync/EmersonLab/Research/Ch3Wildfire/R_Figures/rds/lnu_phylum_long.RDS")

# Prep data frame for plotting
lnu.phylum.map <- merge(lnu.phylum.long, wildfire_asv_map, by="ASV_ID")

# Plot
lnu_phylum <- ggplot(lnu.phylum.map, aes(x=Treatment, y=counts, fill=NewPhylum))+
    labs(x="Treatment", 
       y ="Counts",
       title = "LNU 16S rRNA ASV Profile")+
  geom_bar(stat="identity", position = "fill")+
  scale_fill_manual(values=collapsed_phylum_colors) +
  theme_bw() +
  #theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  facet_wrap(~ Habitat)

lnu_phylum

lnu_phylum <- ggsave("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch3Wildfire/R_Figures/pdf/StackedBar/lnu_phylum.pdf", width = 200, height = 150, units = "mm", dpi = 500)

ggplot(lnu.phylum.map, aes(x=TimePoint, y=counts, fill=NewPhylum))+
    labs(x="TimePoint", 
       y ="Counts",
       title = "LNU 16S rRNA ASV Profile")+
  geom_bar(stat="identity", position = "fill")+
  scale_fill_manual(values=collapsed_phylum_colors) +
  theme_bw() +
  #theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  facet_grid(Habitat ~ Treatment)

ggsave("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch3Wildfire/R_Figures/pdf/StackedBar/lnu_phylum_timepoint.pdf", width = 200, height = 150, units = "mm", dpi = 500)

burn_lnu.phylum.map <- lnu.phylum.map[(lnu.phylum.map$Treatment=="AfterBurn_Wildfire"),]

control_lnu.phylum.map <- lnu.phylum.map[(lnu.phylum.map$Treatment=="AfterBurn_Control"),]


### PUT LINE BETWEEN TIMEPOINTS

ggplot(burn_lnu.phylum.map, aes(x=TimePoint_ID, y=counts, fill=NewPhylum))+
    labs(x="TimePoint", 
       y ="Counts",
       title = "LNU 16S rRNA ASV Profile")+
  geom_bar(stat="identity", position = "fill")+
  scale_fill_manual(values=collapsed_phylum_colors) +
  geom_vline(xintercept = c(4.5, 8.5, 16.5, 24.5), linetype = "solid", color = "grey20", linewidth = 0.15)+
  theme_bw() +
  theme(legend.position = "bottom") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  facet_grid(~Habitat)
ggsave("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch3Wildfire/R_Figures/pdf/StackedBar/prokaryotes/burn_lnu_phylum_timepoint.pdf", width = 300, height = 150, units = "mm", dpi = 500)

ggplot(control_lnu.phylum.map, aes(x=TimePoint_ID, y=counts, fill=NewPhylum))+
    labs(x="TimePoint", 
       y ="Counts",
       title = "LNU 16S rRNA ASV Profile")+
  geom_bar(stat="identity", position = "fill")+
  scale_fill_manual(values=collapsed_phylum_colors) +
  geom_vline(xintercept = c(4.5, 8.5, 16.5), linetype = "solid", color = "grey20", linewidth = 0.15)+
  theme_bw() +
  theme(legend.position = "bottom") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  facet_grid(~Habitat)
ggsave("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch3Wildfire/R_Figures/pdf/StackedBar/prokaryotes/control_lnu_phylum_timepoint.pdf", width = 300, height = 150, units = "mm", dpi = 500)


lnu.phylum.relabd <- lnu.phylum.map %>%
  group_by(ASV_ID) %>%
  mutate(RelAbd = counts / 17700)

lnu.phylum.avg <- lnu.phylum.relabd %>%
  group_by(DaysAfterFire, NewPhylum, Habitat, Treatment) %>%
  dplyr::summarize(MeanRelAbd = mean(RelAbd, na.rm = TRUE))

ggplot(lnu.phylum.avg, aes(x=DaysAfterFire, y=MeanRelAbd, color=NewPhylum, group=NewPhylum))+
    labs(x="Time Since Fire (Days)", 
       y ="Mean Relative Abundance",
       title = "LNU 16S rRNA ASV Profile")+
  geom_line() +
  geom_point() +
  scale_color_manual(values=collapsed_phylum_colors) +
  theme_bw() +
  #theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  facet_grid(Habitat ~ Treatment)
```

# Taxonomy Stacked Bar - Chaparral, Burn vs Control (T3, T4, T5)
```{r}
# add OTU_ID back onto asv.rare
bc.ch.asv.rare$OTU_ID <- rownames(bc.ch.asv.rare)

# Join taxonomy and abundance data together
bc.ch.tax_lnu <- left_join(bc.ch.asv.rare, tax_filtered, by = "OTU_ID")
# make sure asv.rare is in the same order as tax_lnu (return = TRUE)
check_column <- "OTU_ID"
all(bc.ch.asv.rare[[check_column]] == bc.ch.tax_lnu[[check_column]])
unique(bc.ch.tax_lnu$Phylum)

# pivot table and group by Phylum
bc.ch.tax_lnu.long <- pivot_longer(bc.ch.tax_lnu, cols=1:35, names_to = "ASV_ID", values_to = "counts")
bc.ch.lnu.phylum.long <- group_by(bc.ch.tax_lnu.long, Phylum, ASV_ID) %>%
  dplyr::summarize(counts=sum(counts))

# collapse phyla into "other" category if represents less than 2%, which is 354 (rarefied to 17,700)
bc.ch.lnu.phylum.long$Phylum <- as.character(bc.ch.lnu.phylum.long$Phylum)
bc.ch.lnu.phylum.long <- bc.ch.lnu.phylum.long %>%
  group_by(Phylum) %>%
  mutate(NewPhylum = ifelse(all(counts < 354), 'Other', Phylum)) %>%
  ungroup()
# find out which phyla are represented (used for making legend)
unique(bc.ch.lnu.phylum.long$NewPhylum)

bc.ch_map <- bc.ch_map %>%
  mutate(Time_Treatment_ID = paste(TimePoint, "-", Treatment, "-", SubPlot)) %>%
  arrange(Time_Treatment_ID)
# Prep data frame for plotting
bc.ch.lnu.phylum.map <- merge(bc.ch.lnu.phylum.long, bc.ch_map, by="ASV_ID")


ggplot(bc.ch.lnu.phylum.map, aes(x=Time_Treatment_ID, y=counts, fill=NewPhylum))+
    labs(x="TimePoint", 
       y ="Counts",
       title = "LNU - Chaparral - burn vs. control - 16S rRNA ASV Profile")+
   annotate('rect', xmin = -Inf, xmax = 4.5, ymin = -Inf, ymax = Inf, fill = 'lightgray', alpha = 0.5) +
  annotate('rect', xmin = 12.5, xmax = 16.5, ymin = -Inf, ymax = Inf, fill = 'lightgray', alpha = 0.5) +
  annotate('rect', xmin = 24.5, xmax = 28.5, ymin = -Inf, ymax = Inf, fill = 'lightgray', alpha = 0.5) +
  geom_bar(stat="identity", position = "fill")+
  scale_fill_manual(values=collapsed_phylum_colors) +
  scale_x_discrete(labels = bc.ch_map$TimePoint_ID) +
  geom_vline(xintercept = c(12.5, 24.5), linetype = "solid", color = "grey20", linewidth = 0.25)+
  theme_bw() +
  theme(legend.position = "bottom") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
ggsave("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch3Wildfire/R_Figures/pdf/StackedBar/prokaryotes/burn_control_chaparral.pdf", width = 300, height = 150, units = "mm", dpi = 500)
```


# Phylum-level stats - Chaparral - Treatment
```{r}
phylum.asv.stats <- lnu.phylum.map

ch.acido_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Acidobacteriota" & Habitat == "Chaparral")
ch.actino_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Actinobacteriota" & Habitat == "Chaparral")
ch.bact_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Bacteroidota" & Habitat == "Chaparral")
ch.chloro_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Chloroflexota" & Habitat == "Chaparral")
ch.crena_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Crenarchaeota" & Habitat == "Chaparral")
ch.firm_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Firmicutes" & Habitat == "Chaparral")
ch.gemma_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Gemmatimonadota" & Habitat == "Chaparral")
ch.myxo_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Myxococcota" & Habitat == "Chaparral")
ch.planc_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Planctomycetota" & Habitat == "Chaparral")
ch.proteo_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Proteobacteria" & Habitat == "Chaparral")
ch.verru_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Verrucomicrobiota" & Habitat == "Chaparral")

kruskal.test(counts ~ Treatment, ch.acido_phylum.asv)
kruskal.test(counts ~ Treatment, ch.actino_phylum.asv)
kruskal.test(counts ~ Treatment, ch.bact_phylum.asv)
kruskal.test(counts ~ Treatment, ch.chloro_phylum.asv)
kruskal.test(counts ~ Treatment, ch.crena_phylum.asv)
kruskal.test(counts ~ Treatment, ch.firm_phylum.asv)
kruskal.test(counts ~ Treatment, ch.gemma_phylum.asv)
kruskal.test(counts ~ Treatment, ch.myxo_phylum.asv)
kruskal.test(counts ~ Treatment, ch.planc_phylum.asv)
kruskal.test(counts ~ Treatment, ch.proteo_phylum.asv)
kruskal.test(counts ~ Treatment, ch.verru_phylum.asv)
```

# Phylum-level stats - Chaparral - Time Series
```{r}
b.ch.acido_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Acidobacteriota" & Habitat == "Chaparral" & Treatment == "AfterBurn_Wildfire")
b.ch.actino_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Actinobacteriota" & Habitat == "Chaparral" & Treatment == "AfterBurn_Wildfire")
b.ch.bact_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Bacteroidota" & Habitat == "Chaparral" & Treatment == "AfterBurn_Wildfire")
b.ch.chloro_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Chloroflexota" & Habitat == "Chaparral" & Treatment == "AfterBurn_Wildfire")
b.ch.crena_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Crenarchaeota" & Habitat == "Chaparral" & Treatment == "AfterBurn_Wildfire")
b.ch.firm_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Firmicutes" & Habitat == "Chaparral" & Treatment == "AfterBurn_Wildfire")
b.ch.gemma_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Gemmatimonadota" & Habitat == "Chaparral" & Treatment == "AfterBurn_Wildfire")
b.ch.myxo_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Myxococcota" & Habitat == "Chaparral" & Treatment == "AfterBurn_Wildfire")
b.ch.planc_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Planctomycetota" & Habitat == "Chaparral" & Treatment == "AfterBurn_Wildfire")
b.ch.proteo_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Proteobacteria" & Habitat == "Chaparral" & Treatment == "AfterBurn_Wildfire")
b.ch.verru_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Verrucomicrobiota" & Habitat == "Chaparral" & Treatment == "AfterBurn_Wildfire")

kruskal.test(counts ~ TimePoint, b.ch.acido_phylum.asv)
cor.test(b.ch.acido_phylum.asv$TimePoint, b.ch.acido_phylum.asv$counts, method = "spearman")




kruskal.test(counts ~ TimePoint, b.ch.actino_phylum.asv)
kruskal.test(counts ~ TimePoint, b.ch.bact_phylum.asv)
kruskal.test(counts ~ TimePoint, b.ch.chloro_phylum.asv)
kruskal.test(counts ~ TimePoint, b.ch.crena_phylum.asv)
kruskal.test(counts ~ TimePoint, b.ch.firm_phylum.asv)
kruskal.test(counts ~ TimePoint, b.ch.gemma_phylum.asv)
kruskal.test(counts ~ TimePoint, b.ch.myxo_phylum.asv)
kruskal.test(counts ~ TimePoint, b.ch.planc_phylum.asv)
kruskal.test(counts ~ TimePoint, b.ch.proteo_phylum.asv)
kruskal.test(counts ~ TimePoint, b.ch.verru_phylum.asv)
```

# Taxonomy Stacked Bar - Woodland, Burn vs Control (T3, T4, T5)
```{r}
# add OTU_ID back onto asv.rare
bc.wo.asv.rare$OTU_ID <- rownames(bc.wo.asv.rare)

# Join taxonomy and abundance data together
bc.wo.tax_lnu <- left_join(bc.wo.asv.rare, tax_filtered, by = "OTU_ID")
# make sure asv.rare is in the same order as tax_lnu (return = TRUE)
check_column <- "OTU_ID"
all(bc.wo.asv.rare[[check_column]] == bc.wo.tax_lnu[[check_column]])
unique(bc.wo.tax_lnu$Phylum)

# pivot table and group by Phylum
bc.wo.tax_lnu.long <- pivot_longer(bc.wo.tax_lnu, cols=1:36, names_to = "ASV_ID", values_to = "counts")
bc.wo.lnu.phylum.long <- group_by(bc.wo.tax_lnu.long, Phylum, ASV_ID) %>%
  dplyr::summarize(counts=sum(counts))

# collapse phyla into "other" category if represents less than 2%, which is 374 (rarefied to 18,700)
bc.wo.lnu.phylum.long$Phylum <- as.character(bc.wo.lnu.phylum.long$Phylum)
bc.wo.lnu.phylum.long <- bc.wo.lnu.phylum.long %>%
  group_by(Phylum) %>%
  mutate(NewPhylum = ifelse(all(counts < 374), 'Other', Phylum)) %>%
  ungroup()
# find out which phyla are represented (used for making legend)
unique(bc.wo.lnu.phylum.long$NewPhylum)

bc.wo_map <- bc.wo_map %>%
  mutate(Time_Treatment_ID = paste(TimePoint, "-", Treatment, "-", SubPlot)) %>%
  arrange(Time_Treatment_ID)
# Prep data frame for plotting
bc.wo.lnu.phylum.map <- merge(bc.wo.lnu.phylum.long, bc.wo_map, by="ASV_ID")


ggplot(bc.wo.lnu.phylum.map, aes(x=Time_Treatment_ID, y=counts, fill=NewPhylum))+
    labs(x="TimePoint", 
       y ="Counts",
       title = "LNU - Woodland - burn vs. control - 16S rRNA ASV Profile")+
  annotate('rect', xmin = -Inf, xmax = 4.5, ymin = -Inf, ymax = Inf, fill = 'lightgray', alpha = 0.5) +
  annotate('rect', xmin = 12.5, xmax = 16.5, ymin = -Inf, ymax = Inf, fill = 'lightgray', alpha = 0.5) +
  annotate('rect', xmin = 24.5, xmax = 28.5, ymin = -Inf, ymax = Inf, fill = 'lightgray', alpha = 0.5) +
  geom_bar(stat="identity", position = "fill")+
  scale_fill_manual(values=collapsed_phylum_colors) +
  scale_x_discrete(labels = bc.wo_map$TimePoint_ID) +
  geom_vline(xintercept = c(12.5, 24.5), linetype = "solid", color = "grey20", linewidth = 0.25)+
  theme_bw() +
  theme(legend.position = "bottom") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
ggsave("C:/Users/segeo/Box Sync/EmersonLab/Research/Ch3Wildfire/R_Figures/pdf/StackedBar/prokaryotes/burn_control_woodland.pdf", width = 300, height = 150, units = "mm", dpi = 500)
```

# Phylum-level stats - Woodland - Treatment
```{r}
wo.acido_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Acidobacteriota" & Habitat == "Woodland")
wo.actino_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Actinobacteriota" & Habitat == "Woodland")
wo.bact_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Bacteroidota" & Habitat == "Woodland")
wo.chloro_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Chloroflexota" & Habitat == "Woodland")
wo.crena_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Crenarchaeota" & Habitat == "Woodland")
wo.firm_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Firmicutes" & Habitat == "Woodland")
wo.gemma_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Gemmatimonadota" & Habitat == "Woodland")
wo.myxo_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Myxococcota" & Habitat == "Woodland")
wo.planc_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Planctomycetota" & Habitat == "Woodland")
wo.proteo_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Proteobacteria" & Habitat == "Woodland")
wo.verru_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Verrucomicrobiota" & Habitat == "Woodland")

kruskal.test(counts ~ Treatment, wo.acido_phylum.asv)
kruskal.test(counts ~ Treatment, wo.actino_phylum.asv)
kruskal.test(counts ~ Treatment, wo.bact_phylum.asv)
kruskal.test(counts ~ Treatment, wo.chloro_phylum.asv)
kruskal.test(counts ~ Treatment, wo.crena_phylum.asv)
kruskal.test(counts ~ Treatment, wo.firm_phylum.asv)
kruskal.test(counts ~ Treatment, wo.gemma_phylum.asv)
kruskal.test(counts ~ Treatment, wo.myxo_phylum.asv)
kruskal.test(counts ~ Treatment, wo.planc_phylum.asv)
kruskal.test(counts ~ Treatment, wo.proteo_phylum.asv)
kruskal.test(counts ~ Treatment, wo.verru_phylum.asv)
```

# Phylum-level stats - Woodland - Time Series
```{r}
b.wo.acido_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Acidobacteriota" & Habitat == "Woodland" & Treatment == "AfterBurn_Wildfire")
b.wo.actino_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Actinobacteriota" & Habitat == "Woodland" & Treatment == "AfterBurn_Wildfire")
b.wo.bact_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Bacteroidota" & Habitat == "Woodland" & Treatment == "AfterBurn_Wildfire")
b.wo.chloro_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Chloroflexota" & Habitat == "Woodland" & Treatment == "AfterBurn_Wildfire")
b.wo.crena_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Crenarchaeota" & Habitat == "Woodland" & Treatment == "AfterBurn_Wildfire")
b.wo.firm_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Firmicutes" & Habitat == "Woodland" & Treatment == "AfterBurn_Wildfire")
b.wo.gemma_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Gemmatimonadota" & Habitat == "Woodland" & Treatment == "AfterBurn_Wildfire")
b.wo.myxo_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Myxococcota" & Habitat == "Woodland" & Treatment == "AfterBurn_Wildfire")
b.wo.planc_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Planctomycetota" & Habitat == "Woodland" & Treatment == "AfterBurn_Wildfire")
b.wo.proteo_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Proteobacteria" & Habitat == "Woodland" & Treatment == "AfterBurn_Wildfire")
b.wo.verru_phylum.asv <- phylum.asv.stats %>%
  filter(Phylum == "Verrucomicrobiota" & Habitat == "Woodland" & Treatment == "AfterBurn_Wildfire")

kruskal.test(counts ~ TimePoint, b.wo.acido_phylum.asv)
kruskal.test(counts ~ TimePoint, b.wo.actino_phylum.asv)
kruskal.test(counts ~ TimePoint, b.wo.bact_phylum.asv)
kruskal.test(counts ~ TimePoint, b.wo.chloro_phylum.asv)
kruskal.test(counts ~ TimePoint, b.wo.crena_phylum.asv)
kruskal.test(counts ~ TimePoint, b.wo.firm_phylum.asv)
kruskal.test(counts ~ TimePoint, b.wo.gemma_phylum.asv)
kruskal.test(counts ~ TimePoint, b.wo.myxo_phylum.asv)
kruskal.test(counts ~ TimePoint, b.wo.planc_phylum.asv)
kruskal.test(counts ~ TimePoint, b.wo.proteo_phylum.asv)
kruskal.test(counts ~ TimePoint, b.wo.verru_phylum.asv)
```