{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# download 7 runs from 2019 study (Quail Ridge and McLaughlin - Woodland and Chaparral)\n",
    "\n",
    "SRR18906547 -> QR_2_2\n",
    "SRR18906548 -> QR_1_2\n",
    "SRR18906549 -> QR_1_1\n",
    "SRR18906550 -> McL_6_2\n",
    "SRR18906551 -> McL_6_1\n",
    "SRR18906557 -> McL_1_2\n",
    "SRR18906558 -> McL_1_1\n",
    "\n",
    "module load sratoolkit/\n",
    "# configure toolkit: https://github.com/ncbi/sra-tools/wiki/03.-Quick-Toolkit-Configuration\n",
    "vdb-config -i\n",
    "# download files\n",
    "prefetch SRR18906558\n",
    "# extract files\n",
    "fasterq-dump SRR18906558/\n",
    "\n",
    "# move all 2019 study files to natrsv directory\n",
    "mkdir SRA\n",
    "mkdir reads\n",
    "\n",
    "# process reads\n",
    "cd reads\n",
    "\n",
    "# QUALITY FILTERING\n",
    "nano qual_filter.sh\n",
    " \n",
    "-----------------\n",
    "\n",
    "#!/bin/bash\n",
    "#SBATCH -D /home/seuge91/natrsv/reads\n",
    "#SBATCH --job-name=qual_filter\n",
    "#SBATCH --nodes=1\n",
    "#SBATCH --time=04:00:00\n",
    "#SBATCH --ntasks=8\n",
    "#SBATCH --mem=32GB\n",
    "#SBATCH --partition=bmm\n",
    "\n",
    "# for calculating the amount of time the job takes\n",
    "begin=`date +%s`\n",
    "echo $HOSTNAME\n",
    "\n",
    "# loading modules\n",
    "module load bbmap\n",
    "source /home/csantosm/initconda\n",
    "conda activate TRIMMOMATIC\n",
    "\n",
    "# running commands\n",
    "path=${1}\n",
    "sample=${2}\n",
    "\n",
    "cd ${path}\n",
    "\n",
    "trimmomatic PE -threads 8 -phred33 \\\n",
    "  ./reads/raw/${sample}_1.fastq ./reads/raw/${sample}_2.fastq \\\n",
    "  ./reads/trimmed/${sample}_R1_trimmed.fq ./reads/unpaired/${sample}_R1_unpaired.fq \\\n",
    "  ./reads/trimmed/${sample}_R2_trimmed.fq ./reads/unpaired/${sample}_R2_unpaired.fq \\\n",
    "  ILLUMINACLIP:/home/csantosm/databases/TruSeq3-PE.fa:2:30:10 \\\n",
    "  SLIDINGWINDOW:4:30 MINLEN:50\n",
    "\n",
    "  bbduk.sh \\\n",
    "    in1=./reads/trimmed/${sample}_R1_trimmed.fq \\\n",
    "    in2=./reads/trimmed/${sample}_R2_trimmed.fq \\\n",
    "    out1=./reads/rmphix/${sample}_R1_rmphix.fq \\\n",
    "    out2=./reads/rmphix/${sample}_R2_rmphix.fq \\\n",
    "    ref=/home/csantosm/databases/phix174_ill.ref.fa \\\n",
    "    k=31 \\\n",
    "    hdist=1 \\\n",
    "    stats=./reads/stats/${sample}_stats.txt \\\n",
    "    -Xmx20g\n",
    "\n",
    "  bbduk.sh \\\n",
    "    in=./reads/unpaired/${sample}_R1_unpaired.fq \\\n",
    "    out=./reads/rmphix_unpaired/${sample}_R1_rmphix_unpaired.fq \\\n",
    "    ref=/home/csantosm/databases/phix174_ill.ref.fa \\\n",
    "    k=31 \\\n",
    "    hdist=1 \\\n",
    "    stats=./reads/stats/${sample}_R1_stats.txt \\\n",
    "    -Xmx20g\n",
    "\n",
    "  bbduk.sh \\\n",
    "    in=./reads/unpaired/${sample}_R2_unpaired.fq \\\n",
    "    out=./reads/rmphix_unpaired/${sample}_R2_rmphix_unpaired.fq \\\n",
    "    ref=/home/csantosm/databases/phix174_ill.ref.fa \\\n",
    "    k=31 \\\n",
    "    hdist=1 \\\n",
    "    stats=./reads/stats/${sample}_R2_stats.txt \\\n",
    "    -Xmx20g\n",
    "\n",
    "# finished commands\n",
    "\n",
    "# getting end time to calculate time elapsed\n",
    "end=`date +%s`\n",
    "elapsed=`expr $end - $begin`\n",
    "echo Time taken: $elapsed\n",
    "    \n",
    "----------------------------\n",
    "\n",
    "# to run commands\n",
    "\n",
    "bash\n",
    "for sample in $(</home/seuge91/natrsv/sample.txt)\n",
    "do\n",
    "sbatch \\\n",
    "--output=/home/seuge91/natrsv/reads/log/${sample}.qf.log \\\n",
    "--error=/home/seuge91/natrsv/reads/err/${sample}.qf.err \\\n",
    "/home/seuge91/natrsv/reads/qual_filter.sh /home/seuge91/natrsv $sample\n",
    "done\n",
    "\n",
    "----------------------------------------------------------------------------------------------------------------\n",
    "\n",
    "# ASSEMBLY\n",
    "cd ~natrsv/megahit\n",
    "nano megahit.sh\n",
    "\n",
    "----------------------------\n",
    "\n",
    "#!/bin/bash\n",
    "#SBATCH -D /home/seuge91/natrsv/megahit/\n",
    "#SBATCH --job-name=megahit2\n",
    "#SBATCH --nodes=1\n",
    "#SBATCH -t 24:00:00\n",
    "#SBATCH --ntasks=8\n",
    "#SBATCH --partition=high2\n",
    "#SBATCH --mem=50gb\n",
    "\n",
    "# for calculating the amount of time the job takes\n",
    "begin=`date +%s`\n",
    "echo $HOSTNAME\n",
    "\n",
    "# loading modules\n",
    "source /home/csantosm/initconda\n",
    "conda activate MEGAHIT\n",
    "\n",
    "# move to rawreads folder\n",
    "cd /home/seuge91/natrsv/reads/\n",
    "sample=${1}\n",
    "\n",
    "megahit -1 ./rmphix/${sample}_R1_rmphix.fq \\\n",
    "-2 ./rmphix/${sample}_R2_rmphix.fq \\\n",
    "-r ./rmphix_unpaired/${sample}_R1_rmphix_unpaired.fq,./rmphix_unpaired/${sample}_R2_rmphix_unpaired.fq \\\n",
    "-o ../megahit/${sample} \\\n",
    "--out-prefix ${sample} \\\n",
    "--min-contig-len 10000 --presets meta-large \\\n",
    "-t 8 --continue\n",
    "\n",
    "# getting end time to calculate time elapsed\n",
    "end=`date +%s`\n",
    "elapsed=`expr $end - $begin`\n",
    "echo Time taken: $elapsed\n",
    "    \n",
    "----------------------------\n",
    "\n",
    "# to run commands\n",
    "\n",
    "bash\n",
    "for sample in $(</home/seuge91/natrsv/sample.txt)\n",
    "do\n",
    "sbatch \\\n",
    "--output=/home/seuge91/natrsv/megahit/log/${sample}.mh.log \\\n",
    "--error=/home/seuge91/natrsv/megahit/err/${sample}.mh.err \\\n",
    "/home/seuge91/natrsv/megahit/megahit2.sh $sample\n",
    "done\n",
    "\n",
    "# to run samples that were suspended/unfinished\n",
    "# make updated sampleIDs list with samples that need to be resubmitted \n",
    "# I made redo.txt \n",
    "\n",
    "# to run commands\n",
    "\n",
    "bash\n",
    "for sample in $(</home/seuge91/natrsv/redo.txt)\n",
    "do\n",
    "sbatch \\\n",
    "--output=/home/seuge91/natrsv/megahit/log/${sample}.mh.log \\\n",
    "--error=/home/seuge91/natrsv/megahit/err/${sample}.mh.err \\\n",
    "/home/seuge91/natrsv/megahit/megahit2.sh $sample\n",
    "done\n",
    "\n",
    "----------------------------\n",
    "\n",
    "# to count how many contigs assembled in each sample\n",
    "\n",
    "grep -rc \"k127\"\n",
    "\n",
    "----------------------------------------------------------------------------------------------------------------\n",
    "\n",
    "# VIRAL DETECTION\n",
    "\n",
    "cd megahit\n",
    "mkdir contigs\n",
    "mkdir renamed_contigs\n",
    "\n",
    "# move all contig fasta files into one folder\n",
    "cd /home/seuge91/natrsv/megahit\n",
    "mv */*contigs.fa contigs\n",
    "cd contigs\n",
    "\n",
    "# rename contig files so that it shows exactly what sample each contig came from\n",
    "module load bbmap\n",
    "\n",
    "for sample in $(<../../sample.txt)\n",
    "do\n",
    "rename.sh in=${sample}.contigs.fa out=${sample}.renamed.contigs.fa prefix=${sample}_contig_\n",
    "done\n",
    "\n",
    "# it didn't work because java wasnt installed\n",
    "# had to load deprecated version for it to work\n",
    "module load deprecated/java\n",
    "\n",
    "mv *renamed.contigs.fa ../renamed_contigs\n",
    "\n",
    "cd /home/seuge91/natrsv\n",
    "mkdir vibrant\n",
    "cd vibrant\n",
    "mkdir err log\n",
    "\n",
    "# make vibrant script\n",
    "nano vibrant.sh\n",
    "\n",
    "----------------------------\n",
    "\n",
    "#!/bin/bash\n",
    "#SBATCH -D /home/seuge91/natrsv/vibrant/\n",
    "#SBATCH --job-name=vbrnt\n",
    "#SBATCH --nodes=1\n",
    "#SBATCH -t 12:00:00\n",
    "#SBATCH --ntasks=4\n",
    "#SBATCH --partition=high2\n",
    "#SBATCH --mem=50gb\n",
    "\n",
    "# for calculating the amount of time the job takes\n",
    "begin=`date +%s`\n",
    "echo $HOSTNAME\n",
    "\n",
    "#activate personal conda env\n",
    "source /home/csantosm/initconda\n",
    "conda activate VIBRANT\n",
    "\n",
    "# move to main folder\n",
    "cd /home/seuge91/natrsv/\n",
    "sample=${1}\n",
    "\n",
    "VIBRANT_run.py -i ./megahit/renamed_contigs/${sample}.renamed.contigs.fa \\\n",
    "-folder ./vibrant/${sample}_vibrant \\\n",
    "-t 4 -f nucl -virome\n",
    "\n",
    "#getting end time to calculate time elapsed\n",
    "end=`date +%s`\n",
    "elapsed=`expr $end - $begin`\n",
    "echo Time taken: $elapsed\n",
    "        \n",
    "----------------------------\n",
    "\n",
    "# to run commands\n",
    "\n",
    "bash\n",
    "for sample in $(</home/seuge91/natrsv/sample.txt)\n",
    "do\n",
    "sbatch \\\n",
    "--output=/home/seuge91/natrsv/vibrant/log/${sample}.vb.log \\\n",
    "--error=/home/seuge91/natrsv/vibrant/err/${sample}.vb.err \\\n",
    "/home/seuge91/natrsv/vibrant/vibrant.sh $sample\n",
    "done\n",
    "\n",
    "# prepare files for dereplication\n",
    "# in vibrant folder, make new directory for vibrant contigs\n",
    "mkdir vibrant_contigs\n",
    "\n",
    "# copy all contigs to directory\n",
    "cp */*/VIBRANT_phages*/*phages_combined.fna ./vibrant_contigs\n",
    "\n",
    "# in project directory, make drep directory\n",
    "mkdir drep\n",
    "\n",
    "# make subdirectory\n",
    "mkdir all\n",
    "\n",
    "# make folder for split contigs in each subdirectory\n",
    "mkdir split_contigs\n",
    "\n",
    "# move all viral contigs from each sample from vibrant to the folder called \"split_contigs\" ...\n",
    "# ...by first moving\" into vibrant folder and then vibrant_contigs directory\n",
    "cd vibrant\n",
    "cd updated_vibrant_contigs\n",
    "\n",
    "# concatenate all viral contigs into a single file\n",
    "cat *phages_combined.fna > natrsv.vib.contigs.fna\n",
    "\n",
    "# count how many total contigs\n",
    "grep -c contig natrsv.vib.contigs.fna\n",
    "\n",
    "# natrsv = 436 contigs\n",
    "\n",
    "# move combined file to LNU directory... continue on to next cell"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Move LNU predicted vOTUs to a new directory\n",
    "mkdir LNU\n",
    "cd fire/vibrant\n",
    "cp lnu.vib.contigs.fna ../../LNU/\n",
    "cd ~/LNU\n",
    "\n",
    "# starting input files\n",
    "lnu.vib.contigs.fna\n",
    "    # count how many total contigs\n",
    "    grep -c contig lnu.vib.contigs.fna\n",
    "    # 77,433 contigs \n",
    "    grep -c contig natrsv.vib.contigs.fna\n",
    "    # 436 contigs\n",
    "# combined input file\n",
    "    # count how many total contigs\n",
    "    grep -c contig all.vib.contigs.fna\n",
    "    # 77,869 contigs\n",
    "    \n",
    "# make cdhit directory\n",
    "mkdir cdhit\n",
    "\n",
    "# cluster using CD HIT\n",
    "# rename file to use for CD Hit (input.fna)\n",
    "cp all.vib.contigs.fna ./input.fna\n",
    "\n",
    "# create CD HIT script\n",
    "nano cdhitlnu.sh\n",
    "\n",
    "-----------------\n",
    "#!/bin/bash\n",
    "#SBATCH --job-name=cdhit\n",
    "#SBATCH --nodes=1\n",
    "#SBATCH -t 24:00:00\n",
    "#SBATCH --ntasks=10\n",
    "#SBATCH --output=cdhit%j.out\n",
    "#SBATCH --error=cdhit%j.err\n",
    "#SBATCH --partition=bmh\n",
    "\n",
    "module load cdhit\n",
    "cd-hit-est -i input.fna -o clustered_lnu.fna \\\n",
    "-c 0.95 -aS 0.85 -d 0 -M 14000 -T 10\n",
    "-----------------\n",
    "\n",
    "# run CD HIT script for LNU contigs\n",
    "sbatch --output=./log/lnu.log --error=./err/lnu.err cdhitlnu.sh\n",
    "\n",
    "# number of contigs after CD HIT: 77,869 --> 70,135\n",
    "\n",
    "# now run dereplication\n",
    "\n",
    "# take the concatenated file and move it to the appropriate split_contig folder within a drep/\"subset\"/split_contigs\n",
    "mv clustered_lnu.fna ../drep/all/split_contigs\n",
    "\n",
    "# separate the concatenated fasta file into multiple ones\n",
    "# use screen because this will take awhile...\n",
    "screen\n",
    "cd ~/LNU/drep/all/split_contigs\n",
    "awk '/^>/ {OUT=substr($0,2) \".fa\"}; OUT {print >OUT}' *.fna \n",
    "    # if using screen...you can monitor how close it is to done by counting file names\n",
    "        ls -1 | wc -l        \n",
    "# detach from screen: ctrl a + d\n",
    "# to see screens running\n",
    "screen -r\n",
    "# to end screen\n",
    "exit\n",
    "\n",
    "# once done, remove the concatenated file from folder (you can move up a folder)\n",
    "mv clustered_all.fna ../\n",
    "\n",
    "# divide contigs into ~ six batches \n",
    "mkdir batch1 batch2 batch3 batch4\n",
    "\n",
    "# within each batch, make a split_contigs folder\n",
    "cd batch1\n",
    "mkdir split_contigs\n",
    "\n",
    "# because 70,136 is too many for one run of drep, it will run out of memory\n",
    "# each batch needs to be less than 50,000, so divide into 4 batches\n",
    "cd all/split_contigs\n",
    "ls | head -n 17534 | xargs -I {} mv {} ../../batch1/split_contigs\n",
    "ls | head -n 17534 | xargs -I {} mv {} ../../batch2/split_contigs\n",
    "ls | head -n 17534 | xargs -I {} mv {} ../../batch3/split_contigs\n",
    "ls | head -n 17534 | xargs -I {} mv {} ../../batch4/split_contigs\n",
    "\n",
    "# make drep script\n",
    "nano drep.sh\n",
    "-----------------------------------------------------------------------------------------\n",
    "\n",
    "#!/bin/bash\n",
    "#SBATCH -D /home/seuge91/LNU/drep\n",
    "#SBATCH --job-name=drep\n",
    "#SBATCH --nodes=1\n",
    "#SBATCH -t 72:00:00\n",
    "#SBATCH --ntasks=24\n",
    "#SBATCH --mem=900GB\n",
    "#SBATCH --partition=bmh\n",
    "\n",
    "# for calculating the amount of time the job takes\n",
    "begin=`date +%s`\n",
    "echo $HOSTNAME\n",
    "\n",
    "# activate personal conda env\n",
    "source /home/csantosm/initconda\n",
    "conda activate DREP\n",
    "\n",
    "treatment=${1}\n",
    "\n",
    "cd /home/seuge91/LNU/drep/${treatment}\n",
    "\n",
    "dRep dereplicate ./${treatment}_dRep \\\n",
    "-g ./split_contigs/*.fa \\\n",
    "--S_algorithm ANImf \\\n",
    "-sa 0.95 \\\n",
    "-nc 0.85 \\\n",
    "-l 10000 \\\n",
    "-N50W 0 \\\n",
    "-sizeW 1 \\\n",
    "--ignoreGenomeQuality \\\n",
    "--clusterAlg single \\\n",
    "-p 24\n",
    "\n",
    "#getting end time to calculate time elapsed\n",
    "end=`date +%s`\n",
    "elapsed=`expr $end - $begin`\n",
    "echo Time taken: $elapsed\n",
    "    \n",
    "-----------------------------------------------------------------------------------------\n",
    "# run the script giving the variable of the subset folder that has the split_contigs directory\n",
    "sbatch drep.sh batch1\n",
    "sbatch drep.sh batch2\n",
    "sbatch drep.sh batch3\n",
    "sbatch drep.sh batch4\n",
    "\n",
    "-----------------------------------------------------------------------------------------\n",
    "\n",
    "# number of contigs after drep: 70,135 --> ?? \n",
    "# batch1: 17,534 --> 16,858\n",
    "# batch2: 17,534 --> 16,968\n",
    "# batch3: 17,534 --> 16,975\n",
    "# batch4: 17,534 --> 16,969\n",
    "\n",
    "# Ok whatever, I'm just going to use the CD HIT output...\n",
    "\n",
    "# how to download the CD HIT output file\n",
    "scp seuge91@farm.cse.ucdavis.edu:/home/seuge91/blodgett_combined/cdhit/output/clustered_all.fna.clstr /mnt/c/Users/segeo/Downloads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# read mapping\n",
    "\n",
    "# make the directory structure for bowtie\n",
    "mkdir bowtie\n",
    "cd bowtie\n",
    "mkdir err log alignments ref coverm\n",
    "\n",
    "# input will be final CD HIT file\n",
    "drep/all/clustered_lnu.fna\n",
    "\n",
    "# convert this to a .fa file\n",
    "cp clustered_lnu.fna clustered_all.fa\n",
    "\n",
    "# make bowtie reference script\n",
    "nano bowtie_ref.sh\n",
    "\n",
    "# I needed to increase memory to 200GB because such a large dataset...\n",
    "----------------------------\n",
    "\n",
    "#!/bin/bash\n",
    "#SBATCH --job-name=bt2ref\n",
    "#SBATCH --nodes=1\n",
    "#SBATCH -t 12:00:00\n",
    "#SBATCH --ntasks=1\n",
    "#SBATCH --mem=200GB\n",
    "#SBATCH --partition=bmh\n",
    "\n",
    "# for calculating the amount of time the job takes\n",
    "begin=`date +%s`\n",
    "echo $HOSTNAME\n",
    "\n",
    "module load bowtie2\n",
    "\n",
    "cd /home/seuge91/LNU/bowtie/ref/\n",
    "\n",
    "bowtie2-build ../clustered_all.fa all_vibrant_cdhit\n",
    "\n",
    "# getting end time to calculate time elapsed\n",
    "end=`date +%s`\n",
    "elapsed=`expr $end - $begin`\n",
    "echo Time taken: $elapsed\n",
    "    \n",
    "----------------------------\n",
    "\n",
    "# run the script to index the reference database\n",
    "sbatch --output=./log/ref.log --error=./err/ref.err bowtie_ref.sh\n",
    "\n",
    "# map the reads against the database\n",
    "# make mapping script\n",
    "nano bowtie_map.sh\n",
    "\n",
    "----------------------------\n",
    "\n",
    "#!/bin/bash\n",
    "#SBATCH --job-name=bt2map\n",
    "#SBATCH --nodes=1\n",
    "#SBATCH -t 6:00:00\n",
    "#SBATCH --ntasks=48\n",
    "#SBATCH --partition=bmh\n",
    "#SBATCH --mem=50gb\n",
    "\n",
    "# for calculating the amount of time the job takes\n",
    "begin=`date +%s`\n",
    "echo $HOSTNAME\n",
    "\n",
    "#load modules\n",
    "module load bowtie2\n",
    "module load samtools\n",
    "\n",
    "path=/home/seuge91/LNU/\n",
    "sample=${1}\n",
    "\n",
    "cd /home/seuge91/LNU/bowtie/ref/\n",
    "\n",
    "bowtie2 -x all_vibrant_cdhit -p 48 \\\n",
    "-1 ${path}reads/rmphix/${sample}_R1_rmphix.fq.gz \\\n",
    "-2 ${path}reads/rmphix/${sample}_R2_rmphix.fq.gz \\\n",
    "-S ${path}bowtie/alignments/${sample}.vib.sam \\\n",
    "--sensitive\n",
    "\n",
    "cd /home/seuge91/LNU/bowtie/alignments\n",
    "samtools view -F 4 -bS ${sample}.vib.sam | samtools sort > ${sample}.vib.sI.bam\n",
    "samtools index ${sample}.vib.sI.bam\n",
    "\n",
    "rm ${sample}.vib.sam\n",
    "\n",
    "# getting end time to calculate time elapsed\n",
    "end=`date +%s`\n",
    "elapsed=`expr $end - $begin`\n",
    "echo Time taken: $elapsed\n",
    "    \n",
    "----------------------------\n",
    "\n",
    "# edit file names first:\n",
    "for file in *_rmphix_combined.fq.gz; do mv \"$file\" \"${file/_combined/}\"; done\n",
    "\n",
    "# run mapping script\n",
    "for sample in $(<../sample.txt)\n",
    "do\n",
    "  sbatch --output=./log/${sample}.map.log --error=./err/${sample}.map.err bowtie_map.sh $sample\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# generate vOTU table\n",
    "\n",
    "# make coverM script\n",
    "nano coverm.sh\n",
    "\n",
    "----------------------------\n",
    "\n",
    "#!/bin/bash\n",
    "#SBATCH --job-name=coverm\n",
    "#SBATCH --nodes=1\n",
    "#SBATCH -t 10:00:00\n",
    "#SBATCH --ntasks=1\n",
    "#SBATCH --partition=high2\n",
    "\n",
    "# for calculating the amount of time the job takes\n",
    "begin=`date +%s`\n",
    "echo $HOSTNAME\n",
    "\n",
    "## you need to run the command from my coverm folder\n",
    "cd /home/csantosm/software/coverm-x86_64-unknown-linux-musl-0.6.1/\n",
    "\n",
    "path=/home/seuge91/LNU/bowtie/\n",
    "\n",
    "./coverm contig -m trimmed_mean --min-covered-fraction 0.75 -b ${path}/alignments/*.bam > ${path}/coverm/all.good.75.tmean.tsv\n",
    "\n",
    "# getting end time to calculate time elapsed\n",
    "end=`date +%s`\n",
    "elapsed=`expr $end - $begin`\n",
    "echo Time taken: $elapsed\n",
    "    \n",
    "----------------------------\n",
    "\n",
    "# run script\n",
    "sbatch --output=./bowtie/log/coverm.log --output=./bowtie/err/coverm.err coverm.sh\n",
    "\n",
    "# download table\n",
    "\n",
    "scp seuge91@farm.cse.ucdavis.edu:/home/seuge91/LNU/bowtie/coverm/*.tsv /mnt/c/Users/segeo/Downloads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Generating Count Table for Differential Abundance\n",
    "\n",
    "# make covermcount script\n",
    "nano covermcount.sh\n",
    "\n",
    "----------------------------\n",
    "\n",
    "#!/bin/bash\n",
    "#SBATCH --job-name=covermcount\n",
    "#SBATCH --nodes=1\n",
    "#SBATCH -t 10:00:00\n",
    "#SBATCH --ntasks=1\n",
    "#SBATCH --partition=bmh\n",
    "\n",
    "# for calculating the amount of time the job takes\n",
    "begin=`date +%s`\n",
    "echo $HOSTNAME\n",
    "\n",
    "## you need to run the command from my coverm folder\n",
    "cd /home/csantosm/software/coverm-x86_64-unknown-linux-musl-0.6.1/\n",
    "\n",
    "path=/home/seuge91/blodgett_combined/bowtie/\n",
    "\n",
    "./coverm contig -m count -b ${path}/alignments/*.bam > ${path}/coverm/all.count.tsv\n",
    "\n",
    "# getting end time to calculate time elapsed\n",
    "end=`date +%s`\n",
    "elapsed=`expr $end - $begin`\n",
    "echo Time taken: $elapsed\n",
    "        \n",
    "----------------------------\n",
    "    \n",
    "# Download table\n",
    "    \n",
    "scp seuge91@farm.cse.ucdavis.edu:/home/seuge91/heat/bowtie/coverm/*.count.tsv /mnt/c/Users/segeo/Downloads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# host prediction\n",
    "\n",
    "mkdir iphop\n",
    "mkdir err log split_db results\n",
    "nano iphop_split.sh\n",
    "\n",
    "----------------------------\n",
    "\n",
    "#!/bin/bash\n",
    "#SBATCH -D /home/seuge91/LNU/iphop/\n",
    "#SBATCH --job-name=iphop_split\n",
    "#SBATCH --nodes=1\n",
    "#SBATCH -t 1:00:00\n",
    "#SBATCH --ntasks=1\n",
    "#SBATCH --partition=high2\n",
    "\n",
    "# for calculating the amount of time the job takes\n",
    "begin=`date +%s`\n",
    "echo $HOSTNAME\n",
    "\n",
    "source /home/csantosm/initconda\n",
    "conda activate IPHOP\n",
    "\n",
    "set=${1}\n",
    "\n",
    "full_db=/home/seuge91/LNU/bowtie/clustered_all.fa\n",
    "split_db=/home/seuge91/LNU/iphop/split_db\n",
    "\n",
    "iphop split --input_file ${full_db} --split_dir ${split_db}\n",
    "\n",
    "#getting end time to calculate time elapsed\n",
    "end=`date +%s`\n",
    "elapsed=`expr $end - $begin`\n",
    "echo Time taken: $elapsed\n",
    "\n",
    "----------------------------\n",
    "\n",
    "sbatch --output=./log/iphop_split.log --error=./err/iphop_split.err iphop_split.sh\n",
    "\n",
    "nano iphop_predict.sh\n",
    "\n",
    "----------------------------\n",
    "\n",
    "#!/bin/bash\n",
    "#SBATCH -D /home/seuge91/LNU/iphop/\n",
    "#SBATCH --job-name=iphop_predict\n",
    "#SBATCH --nodes=1\n",
    "#SBATCH -t 24:00:00\n",
    "#SBATCH --mem=128GB\n",
    "#SBATCH --ntasks=24\n",
    "#SBATCH --partition=high2\n",
    "\n",
    "# for calculating the amount of time the job takes\n",
    "begin=`date +%s`\n",
    "echo $HOSTNAME\n",
    "\n",
    "source /home/csantosm/initconda\n",
    "conda activate IPHOP\n",
    "\n",
    "batch=${1}\n",
    "\n",
    "iphop_db=/home/csantosm/databases/IPHOP_db/Sept_2021_pub\n",
    "\n",
    "cd /home/seuge91/LNU/iphop\n",
    "\n",
    "iphop predict --fa_file ./split_db/${batch}.fna --out_dir ./results/${batch} --db_dir ${iphop_db} --num_threads 24\n",
    "\n",
    "#getting end time to calculate time elapsed\n",
    "end=`date +%s`\n",
    "elapsed=`expr $end - $begin`\n",
    "echo Time taken: $elapsed\n",
    "    \n",
    "----------------------------\n",
    "\n",
    "batches=\"batch_00000 batch_00009 batch_00018 batch_00027 batch_00036 batch_00045 batch_00054 batch_00063 batch_00001 batch_00010 batch_00019 batch_00028 batch_00037 batch_00046 batch_00055 batch_00064 batch_00002 batch_00011 batch_00020 batch_00029 batch_00038 batch_00047 batch_00056 batch_00065 batch_00003 batch_00012 batch_00021 batch_00030 batch_00039 batch_00048 batch_00057 batch_00066 batch_00004 batch_00013 batch_00022 batch_00031 batch_00040 batch_00049 batch_00058 batch_00067 batch_00005 batch_00014 batch_00023 batch_00032 batch_00041 batch_00050 batch_00059 batch_00068 batch_00006 batch_00015 batch_00024 batch_00033 batch_00042 batch_00051 batch_00060 batch_00069 batch_00007 batch_00016 batch_00025 batch_00034 batch_00043 batch_00052 batch_00061 batch_00070 batch_00008 batch_00017 batch_00026 batch_00035 batch_00044 batch_00053 batch_00062\"\n",
    "for batch in $batches\n",
    "do\n",
    "sbatch --output=./log/${batch}.log --error=./err/${batch}.err iphop_predict.sh $batch\n",
    "done\n",
    "\n",
    "\n",
    "# others that didn't finish or whatever...\n",
    "\n",
    "batches=\"batch_00087 batch_00077 batch_00073\"\n",
    "for batch in $batches\n",
    "do\n",
    "sbatch --output=./log/${batch}.log --error=./err/${batch}.err iphop_predict.sh $batch\n",
    "done\n",
    "\n",
    "# combine results from batches\n",
    "# get header from one of the batches\n",
    "cd results\n",
    "cd batch_00000\n",
    "head -n 1 Host_prediction_to_genome_m90.csv > all_Host_prediction_to_genome_m90.csv\n",
    "head -n 1 Host_prediction_to_genus_m90.csv > all_Host_prediction_to_genus_m90.csv\n",
    "head -n 10 Detailed_output_by_tool.csv > all_Detailed_output_by_tool.csv\n",
    "# move files up a folder\n",
    "mv all_Host_prediction_to_genome_m90.csv ..\n",
    "mv all_Host_prediction_to_genus_m90.csv ..\n",
    "mv all_Detailed_output_by_tool.csv ..\n",
    "# get body of each batch and concatenate to main file\n",
    "cd ../\n",
    "tail -n +2 -q batch_*/Host_prediction_to_genome_m90.csv >> all_Host_prediction_to_genome_m90.csv\n",
    "tail -n +2 -q batch_*/Host_prediction_to_genus_m90.csv >> all_Host_prediction_to_genus_m90.csv\n",
    "tail -n +11 batch_*/Detailed_output_by_tool.csv >> all_Detailed_output_by_tool.csv\n",
    "\n",
    "# download files\n",
    "scp seuge91@farm.cse.ucdavis.edu:/home/seuge91/blodgett_combined/iphop/results/*.csv /mnt/c/Users/segeo/Downloads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## DID NOT NEED TO DO THIS...\n",
    "\n",
    "# PIGEON\n",
    "# download PIGEON database, be sure to be logged out\n",
    "\n",
    "rsync -r --progress /mnt/c/Users/segeo/Downloads/PIGEON.tar.gz seuge91@farm:~/\n",
    "    \n",
    "tar -xvzf PIGEON.tar.gz\n",
    "\n",
    "# How many contigs in PIGEON.fa?\n",
    "grep \">\" PIGEON.fa | wc -l\n",
    "    # 515,763 - correct!\n",
    "\n",
    "# add my contigs from my experiments to PIGEON.fa\n",
    "cat blodgett_PIGEON.fa >> PIGEON.fa   \n",
    "# changed PIGEON.fa to PIGEON3.0.fa\n",
    "\n",
    "# now, I need to decide to do vcontact or read mapping...\n",
    "\n",
    "# read mapping\n",
    "# add LNU contigs to PIGEON\n",
    "cat clustered_all.fa >> ../PIGEON3.0.fa\n",
    "mv PIGEON3.0.fa PIGEON3.0_LNU.fa\n",
    "nano cdhitPIGEON.sh\n",
    "\n",
    "----------------------------\n",
    "\n",
    "#!/bin/bash\n",
    "#SBATCH --job-name=cdhit\n",
    "#SBATCH --nodes=1\n",
    "#SBATCH -t 48:00:00\n",
    "#SBATCH --ntasks=10\n",
    "#SBATCH --output=cdhit%j.out\n",
    "#SBATCH --error=cdhit%j.err\n",
    "#SBATCH --partition=high2\n",
    "#SBATCH --mem=50gb\n",
    "\n",
    "module load cdhit\n",
    "cd-hit-est -i PIGEON3.0_LNU.fa -o clustered_PIGEON_lnu.fna \\\n",
    "-c 0.95 -aS 0.85 -M 14000 -T 10\n",
    "\n",
    "----------------------------\n",
    "\n",
    "sbatch --output=./log/cdhit_pigeon_lnu.log --error=./err/cdhit_pigeon_lnu.err cdhitPIGEON.sh\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# vcontact\n",
    "\n",
    "# make vcontact directory for vibrant contig proteins!!\n",
    "mkdir vcontact\n",
    "\n",
    "# within vcontact directory, make more directories\n",
    "mkdir err log\n",
    "\n",
    "# copy CD-Hit ouput file into vcontact directory\n",
    "cp clustered_lnu.fna /../../vcontact\n",
    "\n",
    "# predict proteins \n",
    "#-i flag means input file, all contigs from CD-HIT \n",
    "#-a flag means output file that is protein, needs to be named with end of \".faa\" \n",
    " \n",
    "source /home/csantosm/initconda \n",
    "conda activate PRODIGAL \n",
    "prodigal -i clustered_lnu.fna -a votus.prodigal.faa -p meta\n",
    "\n",
    "\n",
    "# create gene2genome.csv file\n",
    "# make script\n",
    "nano gtg.sh\n",
    "\n",
    "----------------------------\n",
    "\n",
    "#!/bin/bash\n",
    "#SBATCH -D /home/seuge91/heat/scripts/\n",
    "#SBATCH --job-name=g2g\n",
    "#SBATCH --nodes=1\n",
    "#SBATCH -t 1:00:00\n",
    "#SBATCH --partition=bmm\n",
    "\n",
    "# for calculating the amount of time the job takes\n",
    "begin=`date +%s`\n",
    "echo $HOSTNAME\n",
    "\n",
    "source /home/csantosm/initconda\n",
    "conda activate VCONTACT2\n",
    "\n",
    "cd /home/seuge91/heat/vcontact/\n",
    "\n",
    "vcontact2_gene2genome -p ./votus.prodigal.faa  \\\n",
    "-o ./votus.gene2genome.csv \\\n",
    "-s Prodigal-FAA\n",
    "\n",
    "# getting end time to calculate time elapsed\n",
    "end=`date +%s`\n",
    "elapsed=`expr $end - $begin`\n",
    "echo Time taken: $elapsed\n",
    "\n",
    "----------------------------\n",
    "\n",
    "# run script\n",
    "sbatch --output=../vcontact/log/g2g.log --error=../vcontact/err/g2g.err g2g.sh\n",
    "\n",
    "# create vcontact script\n",
    "cd /home/seuge91/heat/scripts/\n",
    "nano vcontact\n",
    "\n",
    "----------------------------\n",
    "\n",
    "#!/bin/bash\n",
    "#SBATCH -D /home/seuge91/heat/scripts/\n",
    "#SBATCH --job-name=vcontact\n",
    "#SBATCH --nodes=1\n",
    "#SBATCH -t 48:00:00\n",
    "#SBATCH --ntasks=16\n",
    "#SBATCH --partition=bmm\n",
    "\n",
    "# for calculating the amount of time the job takes\n",
    "begin=`date +%s`\n",
    "echo $HOSTNAME\n",
    "\n",
    "source /home/csantosm/initconda\n",
    "conda activate VCONTACT2\n",
    "module load deprecated/java\n",
    "\n",
    "cd /home/seuge91/heat/vcontact/\n",
    "\n",
    "vcontact2 --raw-proteins votus.prodigal.faa \\\n",
    "--rel-mode 'Diamond' \\\n",
    "--db 'ProkaryoticViralRefSeq85-Merged' \\\n",
    "--proteins-fp votus.gene2genome.csv \\\n",
    "--pcs-mode MCL \\\n",
    "--vcs-mode ClusterONE \\\n",
    "--threads 16 \\\n",
    "--c1-bin /home/csantosm/miniconda3/bin/cluster_one-1.0.jar \\\n",
    "--output-dir vcontact_out\n",
    "\n",
    "# getting end time to calculate time elapsed\n",
    "end=`date +%s`\n",
    "elapsed=`expr $end - $begin`\n",
    "echo Time taken: $elapsed\n",
    "    \n",
    "----------------------------\n",
    "\n",
    "# run script\n",
    "sbatch --output=../vcontact/log/vc2.log --error=../vcontact/err/vc2.err vcontact.sh\n",
    "\n",
    "# download files\n",
    "scp seuge91@farm.cse.ucdavis.edu:/home/seuge91/heat/vcontact/vcontact_out/genome_by_genome_overview.csv /mnt/c/Users/segeo/Downloads\n",
    "scp seuge91@farm.cse.ucdavis.edu:/home/seuge91/heat/vcontact/vcontact_out/c1.ntw /mnt/c/Users/segeo/Downloads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# map back all reads to PIGEON database\n",
    "\n",
    "# bow tie reference database in Annie's folder\n",
    "/home/amhorst/PIGEON/bowtie/\n",
    "\n",
    "220711_pigeon\n",
    "\n",
    "# map the reads against the database\n",
    "# make mapping script\n",
    "nano bowtie_map.sh\n",
    "\n",
    "----------------------------\n",
    "\n",
    "#!/bin/bash\n",
    "#SBATCH --job-name=bt2map\n",
    "#SBATCH --nodes=1\n",
    "#SBATCH -t 6:00:00\n",
    "#SBATCH --ntasks=48\n",
    "#SBATCH --partition=high2\n",
    "#SBATCH --mem=50gb\n",
    "\n",
    "# for calculating the amount of time the job takes\n",
    "begin=`date +%s`\n",
    "echo $HOSTNAME\n",
    "\n",
    "#load modules\n",
    "module load bowtie2\n",
    "module load samtools\n",
    "\n",
    "path=/home/seuge91/LNU/\n",
    "sample=${1}\n",
    "\n",
    "cd /home/amhorst/PIGEON/bowtie/\n",
    "\n",
    "bowtie2 -x 220711_pigeon -p 48 \\\n",
    "-1 ${path}reads/rmphix/${sample}_R1_rmphix.fq.gz \\\n",
    "-2 ${path}reads/rmphix/${sample}_R2_rmphix.fq.gz \\\n",
    "-S ${path}bowtie_PIGEON/alignments/${sample}.vib.sam \\\n",
    "--sensitive\n",
    "\n",
    "cd /home/seuge91/LNU/bowtie_PIGEON/alignments\n",
    "samtools view -F 4 -bS ${sample}.vib.sam | samtools sort > ${sample}.vib.sI.bam\n",
    "samtools index ${sample}.vib.sI.bam\n",
    "\n",
    "rm ${sample}.vib.sam\n",
    "\n",
    "# getting end time to calculate time elapsed\n",
    "end=`date +%s`\n",
    "elapsed=`expr $end - $begin`\n",
    "echo Time taken: $elapsed\n",
    "    \n",
    "----------------------------\n",
    "\n",
    "# run mapping script\n",
    "for sample in $(<../sample.txt)\n",
    "do\n",
    "  sbatch --output=./log/${sample}.map.log --error=./err/${sample}.map.err bowtie_map.sh $sample\n",
    "done\n",
    "\n",
    "----------------------------------------------------------------------------------------------------------------\n",
    "\n",
    "# map all reads to blodgett\n",
    "\n",
    "# bow tie reference database in blodgett_combined folder\n",
    "/home/seuge91/blodgett_combined/bowtie/\n",
    "\n",
    "220711_pigeon\n",
    "\n",
    "# map the reads against the database\n",
    "# make mapping script\n",
    "nano bowtie_map.sh\n",
    "\n",
    "----------------------------\n",
    "\n",
    "#!/bin/bash\n",
    "#SBATCH --job-name=bt2map\n",
    "#SBATCH --nodes=1\n",
    "#SBATCH -t 6:00:00\n",
    "#SBATCH --ntasks=48\n",
    "#SBATCH --partition=high2\n",
    "#SBATCH --mem=50gb\n",
    "\n",
    "# for calculating the amount of time the job takes\n",
    "begin=`date +%s`\n",
    "echo $HOSTNAME\n",
    "\n",
    "#load modules\n",
    "module load bowtie2\n",
    "module load samtools\n",
    "\n",
    "path=/home/seuge91/LNU/\n",
    "sample=${1}\n",
    "\n",
    "cd /home/seuge91/blodgett_combined/bowtie/ref\n",
    "\n",
    "bowtie2 -x all_vibrant_cdhit -p 48 \\\n",
    "-1 ${path}reads/rmphix/${sample}_R1_rmphix.fq.gz \\\n",
    "-2 ${path}reads/rmphix/${sample}_R2_rmphix.fq.gz \\\n",
    "-S ${path}bowtie_blodgett/alignments/${sample}.vib.sam \\\n",
    "--sensitive\n",
    "\n",
    "cd /home/seuge91/LNU/bowtie_blodgett/alignments\n",
    "samtools view -F 4 -bS ${sample}.vib.sam | samtools sort > ${sample}.vib.sI.bam\n",
    "samtools index ${sample}.vib.sI.bam\n",
    "\n",
    "rm ${sample}.vib.sam\n",
    "\n",
    "# getting end time to calculate time elapsed\n",
    "end=`date +%s`\n",
    "elapsed=`expr $end - $begin`\n",
    "echo Time taken: $elapsed\n",
    "    \n",
    "----------------------------\n",
    "\n",
    "# run mapping script\n",
    "for sample in $(<../sample.txt)\n",
    "do\n",
    "  sbatch --output=./log/${sample}.map.log --error=./err/${sample}.map.err bowtie_map.sh $sample\n",
    "done\n",
    "\n",
    "# coverage tables\n",
    "# fish for contigs that were found in my dataset using bbmap\n",
    "# anything present - put it all in one fasta file\n",
    "\n",
    "\n",
    "# then cluster with LNU dataset using CD HIT\n",
    "\n",
    "\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
